src/tests/ftest/control/config_generate_run.py

```diff
@@ -0,0 +1,121 @@
+
+        # Call dmg config generate. AP is always the first server host.
+        server_host = self.hostlist_servers[0]
+        generated_yaml = self.get_dmg_command().config_generate(
+            access_points=server_host, num_engines=num_engines,
+            min_ssds=min_ssds, net_class=net_class)
+
+        # Create a temporary file in self.test_dir and write the generated
+        # config.
+        temp_file_path = os.path.join(self.test_dir, "temp_server.yml")
+        try:
+            with open(temp_file_path, 'w') as write_file:
+                yaml.dump(generated_yaml, write_file, default_flow_style=False)
+        except Exception as error:
+            raise CommandFailure(
+                "Error writing the yaml file! {}: {}".format(
+                    temp_file_path, error)) from error
+
+        # Copy the config from temp dir to /etc/daos of the server node.
+        default_server_config = get_default_config_file("server")
+        try:
+            distribute_files(
+                self.hostlist_servers, temp_file_path, default_server_config,
+                verbose=False, sudo=True)
+        except DaosTestError as error:
+            raise CommandFailure(
+                "ERROR: Copying yaml configuration file to {}: "
+                "{}".format(server_host, error)) from error
+
+        # Stop and restart daos_server. self.start_server_managers() has the
+        # server startup check built into it, so if there's something wrong,
+        # it'll throw an error.
+        self.log.debug("## Stopping servers")
+        self.stop_servers()
+
+        # We don't need agent for this test. However, when we stop the server,
+        # agent is also stopped. Then the harness checks that the agent is
+        # running during the teardown. If agent isn't running at that point, it
+        # would cause an error, so start it here.
+        self.log.debug("## Restarting agents")
+        self.start_agent_managers()
+
+        # Before restarting daos_server, we need to clear SCM. Unmount the mount
+        # point, wipefs the disks, etc. This clearing step is built into the
+        # server start steps. It'll look at the engine_params of the
+        # server_manager and clear the SCM set there, so we need to overwrite it
+        # before starting. Set the values from the generated config.
+        self.log.debug("## Resetting engine_params")
+        self.server_managers[0].manager.job.yaml.engine_params = []
+        engines = generated_yaml["engines"]
+        for i, engine in enumerate(engines):
+            self.log.debug("## engine {}".format(i))
+            per_engine_yaml_parameters =\
+                DaosServerYamlParameters.PerEngineYamlParameters(i)
+            self.log.debug("## scm_mount = {}".format(engine["scm_mount"]))
+            per_engine_yaml_parameters.scm_mount.update(engine["scm_mount"])
+            self.log.debug("## scm_class = {}".format(engine["scm_class"]))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions